src/rules/prefer-native-locators.ts

```diff
@@ -0,0 +1,112 @@
+import { AST } from 'eslint'
+import { getStringValue, isPageMethod } from '../utils/ast'
+import { createRule } from '../utils/createRule'
+
+type Pattern = {
+  messageId: string
+  pattern: RegExp
+  replacement: string
+}
+
+export default createRule({
+  create(context) {
+    const { testIdAttribute } = {
+      testIdAttribute: 'data-testid',
+      ...((context.options?.[0] as Record<string, unknown>) ?? {}),
+    }
+
+    const patterns: Array<Pattern> = [
+      {
+        messageId: 'unexpectedLabelQuery',
+        pattern: /^\[aria-label=['"]?(.+?)['"]?\]$/,
```

Comment:
Don't do the entire pattern seperate for each, just the part that is unique (e.g. `role`, `placeholder`).

Reply:
I moved this into a separate function that loops over the data array and compiles the pattern using the same regular expression. Let me know if you have something else in mind.